tests/unit/types/arrays/documentarray/test_documentarray.py

```diff
@@ -344,3 +344,12 @@ def test_none_extend():
     da = DocumentArray([Document() for _ in range(100)])
     da.extend(None)
     assert len(da) == 100
+
+
+def test_lazy_index_map():
+    da = DocumentArray([Document() for _ in range(100)])
+    assert da._id_to_index is None
+
+    # build index map
+    assert len(da._index_map.keys()) == 100
```

Comment:
I would change the test to show that the `_index_map` is not `None` after accessing by `id`

Reply:
done